### PR TITLE
Android build fixes

### PR DESCRIPTION
**Pull request by phire**
These are all the changes I needed to make while trying to get FEX built under termux on my android phone.

I've put this porting effort on hold for now, as NDK r21 doesn't implement `std::filesystem` and termux are still working on updating to r22. 

Apart from `std::filesystem` issues, I manged to get FEXCore building. The LinuxEmulation layer is going to need some more work. 

I have another PR coming which allows building and running IRLoader without LinuxEmulation (which it doesn't use).

### CODE REVIEW

File: Source/Tests/CMakeLists.txt [hunk 1]
```diff
@@ -1,78 +1,21 @@
...
-  LinuxSyscalls/Syscalls/Signals.cpp
-  LinuxSyscalls/Syscalls/Socket.cpp
-  LinuxSyscalls/Syscalls/Thread.cpp
-  LinuxSyscalls/Syscalls/Time.cpp
-  LinuxSyscalls/Syscalls/Timer.cpp
-  LinuxSyscalls/Syscalls/NotImplemented.cpp
-  LinuxSyscalls/Syscalls/Stubs.cpp
-  )
-set(LIBS FEXCore Common CommonCore pthread)
-set(NAME FEXLoader)
```
Comment (phire) [inline Source/Tests/CMakeLists.txt:60]: I removed all the ${NAME} stuff because it made this file very hard to follow and know what ${NAME} was referencing at any-one point.

File: External/FEXCore/Source/Interface/Core/Interpreter/InterpreterOps.cpp [hunk 2]
```diff
@@ -397,13 +398,19 @@ Res GetSrc(void* SSAData, IR::OrderedNodeWrapper Src) {
 [[noreturn]]
 static void StopThread(FEXCore::Core::InternalThreadState *Thread) {
   Thread->CTX->StopThread(Thread);
-  std::unexpected();
+
+  // If the StopThread signal is delayed for any reason, spin forever
+  for (;;)
```
Comment (skmp) [inline External/FEXCore/Source/Interface/Core/Interpreter/InterpreterOps.cpp:403]: How about `LogMan::Msg::A` here?

File: External/FEXCore/Source/Utils/ELFLoader.cpp [hunk 3]
```diff
@@ -412,7 +412,7 @@ void ELFContainer::CalculateSymbols() {
       uint64_t offset = SymTabHeader->sh_offset + i * SymTabHeader->sh_entsize;
       Elf32_Sym const *Symbol =
           reinterpret_cast<Elf32_Sym const *>(&RawFile.at(offset));
-      if (ELF32_ST_VISIBILITY(Symbol->st_other) != STV_HIDDEN &&
+      if (Symbol->st_other & 0x3 != STV_HIDDEN &&
```
Comment (skmp) [inline External/FEXCore/Source/Utils/ELFLoader.cpp:415]: Why was the macro expanded here? It's much cleaner to have it wrapped in a macro

**Review by skmp — CHANGES_REQUESTED** [review]
Looks overall good to me, apart from the noted comments. Probably @Sonicadvance1 should sign off this as well as i'm not too familiar with our cmake files

File: External/FEXCore/Source/Utils/ELFLoader.cpp (hunk 3 above)
Comment (phire) [reply External/FEXCore/Source/Utils/ELFLoader.cpp:415]: The macro doesn't exist on android's version of `elf.h`

Looks like it's a GNU libc extension to the elf.h library.

Comment (Sonicadvance1) [reply External/FEXCore/Source/Utils/ELFLoader.cpp:415]: Might be nice to define our own macro or lambda for this then :)

File: External/FEXCore/Source/Interface/Core/Interpreter/InterpreterOps.cpp (hunk 2 above)
Comment (phire) [reply External/FEXCore/Source/Interface/Core/Interpreter/InterpreterOps.cpp:403]: I'm really not sure that tgkill has any kind of synchronicity guarantee for doing a self kill. 

It might be somewhat common to reach here. 

I'm working on another PR which just stops the thread synchronously.

File: External/FEXCore/Source/Utils/ELFLoader.cpp (hunk 3 above)
Comment (phire) [reply External/FEXCore/Source/Utils/ELFLoader.cpp:415]: Also, this is wrong. Need brackets around `Symbol->st_other & 0x3`

File: Source/Tests/CMakeLists.txt (hunk 1 above)
Comment (Sonicadvance1) [reply Source/Tests/CMakeLists.txt:60]: Reasonable enough

File: External/FEXCore/Source/Interface/Core/Interpreter/InterpreterOps.cpp (hunk 2 above)
Comment (Sonicadvance1) [reply External/FEXCore/Source/Interface/Core/Interpreter/InterpreterOps.cpp:403]: tgkill to itself will immediately switch the thread to the signal handler. It's abused in the Linux syscall handler for pending signal handling.

Comment (phire) [reply External/FEXCore/Source/Interface/Core/Interpreter/InterpreterOps.cpp:403]: Hmm... I checked the linux source code and couldn't really see a code path which guaranteed that.

File: External/FEXCore/Source/Utils/ELFLoader.cpp (hunk 3 above)
Comment (phire) [reply External/FEXCore/Source/Utils/ELFLoader.cpp]: Ok, I'm now redefining the macro if it's not defined.

File: External/FEXCore/Source/Utils/ELFLoader.cpp [hunk 4]
```diff
@@ -435,7 +435,7 @@ void ELFContainer::CalculateSymbols() {
       uint64_t offset = DynSymTabHeader->sh_offset + i * DynSymTabHeader->sh_entsize;
       Elf32_Sym const *Symbol =
           reinterpret_cast<Elf32_Sym const *>(&RawFile.at(offset));
-      if (ELF32_ST_VISIBILITY(Symbol->st_other) != STV_HIDDEN &&
+      if (ELF64_ST_VISIBILITY(Symbol->st_other) & 0x3 != STV_HIDDEN &&
```
Comment (Sonicadvance1) [inline External/FEXCore/Source/Utils/ELFLoader.cpp:438]: Name flipped from other one and oops on still doing & 3?

Comment (phire) [reply External/FEXCore/Source/Utils/ELFLoader.cpp:438]: oops

**Review by Sonicadvance1 — APPROVED** [review]

**Review by Sonicadvance1 — APPROVED** [review]

**Review by skmp — APPROVED** [review]
